### PR TITLE
create snapshotid resource

### PR DESCRIPTION
**Pull request by xh3b4sd**

### CODE REVIEW

File: service/controller/resource/snapshotid/create.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+				},
+			},
+			{
+				Name: aws.String(fmt.Sprintf("tag:%s", key.TagSnapshot)),
+				Values: []*string{
+					aws.String(key.HAMasterSnapshotIDValue),
+				},
+			},
+		},
+	}
```
Comment (lachlanh) [inline service/controller/resource/snapshotid/create.go]: filter on the cluster id snapshot tags

File: service/controller/resource/snapshotid/create.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	count := len(o.Snapshots)
+	if count == 0 {
+		return microerror.Maskf(notExistsError, "snapshot")
+	}
+	if count != 1 {
+		return microerror.Maskf(executionFailedError, "expected one snapshot, got %d", count)
+	}
+
+	cc.Status.TenantCluster.MasterInstance.EtcdVolumeSnapshotID = key.HAMasterSnapshotIDValue
```
Comment (lachlanh) [inline service/controller/resource/snapshotid/create.go]: if we find the ha master snapshot, store it in the controller context, other wise errors. @xh3b4sd is this the right place to be storing it in the controller context ?

File: service/controller/resource/snapshotid/error.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+var invalidConfigError = &microerror.Error{
+	Kind: "invalidConfigError",
+}
+
+var notExistsError = &microerror.Error{
+	Kind: "notExistsError",
+}
+
+var executionFailedError = &microerror.Error{
```
Comment (xh3b4sd) [inline service/controller/resource/snapshotid/error.go]: Let's keep that on top with the comment. Check other packages how they do it.

File: service/controller/resource/snapshotid/create.go (hunk 2 above)
Comment (xh3b4sd) [reply service/controller/resource/snapshotid/create.go]: `key.HAMasterSnapshotIDValue` is not the snapshot ID. We need the real ID so we can create a real volume from it.

Comment (lachlanh) [reply service/controller/resource/snapshotid/create.go]: yup, that was not the right thing to store. should be better now.

File: service/controller/resource/snapshotid/error.go (hunk 3 above)
Comment (lachlanh) [reply service/controller/resource/snapshotid/error.go]: should look the same as tccp/error.

**Pull request merged**